src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,196 @@
+package duke;
+
+import java.io.StreamCorruptedException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.Date;
+import java.text.SimpleDateFormat;
+
+public class Parser {
+    public static Integer parseInt(String str) {
+        try {
+            return Integer.parseInt(str);
+        } catch (NumberFormatException e) {
+            throw new CommandInvalidArgumentFormatException();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public static Date parseDate(String dateStr)
+            throws ParseException {
+        SimpleDateFormat format = new SimpleDateFormat("dd-MM-YYYY");
+        Date date = format.parse(dateStr);
+        return date;
+    }
+
+    /**
+     * The parser is able to parse input from file
+     */
+    public static ArrayList<Task> parseFile(String string)
+            throws StreamCorruptedException, ParseException {
+        Scanner scanner = new Scanner(string);
+        ArrayList tasks = new ArrayList<>();
```

Comment:
I think defining the type of a generic class e.g. `ArrayList<Task>`  here would be better.